src/test/kotlin/com/terraformation/backend/accelerator/db/ParticipantProjectSpeciesStoreTest.kt

```diff
@@ -265,4 +265,102 @@ class ParticipantProjectSpeciesStoreTest : DatabaseTest(), RunsAsUser {
       }
     }
   }
+
+  @Nested
+  inner class Delete {
+    @Test
+    fun `deletes the supplied list of entities by ID`() {
+      val participantId = insertParticipant()
+      val projectId = insertProject(participantId = participantId)
+      val speciesId1 = insertSpecies()
+      val speciesId2 = insertSpecies()
+      val speciesId3 = insertSpecies()
+      val participantProjectSpeciesId1 =
+          insertParticipantProjectSpecies(projectId = projectId, speciesId = speciesId1)
+      val participantProjectSpeciesId2 =
+          insertParticipantProjectSpecies(projectId = projectId, speciesId = speciesId2)
+      val participantProjectSpeciesId3 =
+          insertParticipantProjectSpecies(projectId = projectId, speciesId = speciesId3)
+
+      every { user.canDeleteParticipantProjectSpecies(any()) } returns true
+
+      store.deleteMany(setOf(participantProjectSpeciesId1, participantProjectSpeciesId2))
+
+      assertEquals(
+          listOf(
+              ParticipantProjectSpeciesRow(
+                  feedback = null,
+                  id = participantProjectSpeciesId3,
+                  projectId = projectId,
+                  rationale = null,
+                  speciesId = speciesId3,
+                  submissionStatusId = SubmissionStatus.NotSubmitted)),
+          participantProjectSpeciesDao.findAll())
+    }
+
+    @Test
+    fun `throws exception if an entry does not exist`() {
+      val participantId = insertParticipant()
+      val projectId = insertProject(participantId = participantId)
+      val speciesId = insertSpecies()
+      val participantProjectSpeciesId =
+          insertParticipantProjectSpecies(projectId = projectId, speciesId = speciesId)
+
+      every { user.canDeleteParticipantProjectSpecies(any()) } returns true
+
+      assertThrows<ParticipantProjectSpeciesSetNotFoundException> {
```

Comment:
This test could do a more precise assertion on the exception to verify that its list of IDs is as expected.

If you do that, it might make sense to make the set of IDs a `val` property of the exception, so it gets retained after the exception object is constructed the test can look at the IDs directly rather than having to rely on the exception message being formatted a certain way.

Reply:
In case you haven't run across this pattern before, it'd look something like
```kotlin
try {
  store.deleteMany(...)
  fail("Expected exception")
} catch (e: ParticipantProjectSpeciesSetNotFoundException) {
  assertEquals(setOf(ParticipantProjectSpeciesId(9999)), e.ids, "IDs that couldn't be deleted")
}
```

Reply:
Making the assertion more precise shows that the delete really wasn't pulling in the non-deleted IDs!